runtime-modules/storage/src/benchmarking.rs

```diff
@@ -1,32 +1,1376 @@
+        apply_on_opening_helper::<T, I>(applicant_account_id, applicant_member_id, &opening_id);
+
+    (opening_id, application_id)
+}
+
+fn add_opening_helper<T: Trait + working_group::Trait<I>, I: Instance>(
+    add_opening_origin: &T::Origin,
+    job_opening_type: &OpeningType,
+) -> OpeningId {
+    working_group::Module::<T, I>::add_opening(
+        add_opening_origin.clone(),
+        vec![],
+        *job_opening_type,
+        StakePolicy {
+            stake_amount: <T as working_group::Trait<I>>::MinimumApplicationStake::get(),
+            leaving_unstaking_period: <T as working_group::Trait<I>>::MinUnstakingPeriodLimit::get(
+            ) + One::one(),
+        },
+        Some(One::one()),
+    )
+    .unwrap();
+
+    let opening_id = working_group::Module::<T, I>::next_opening_id() - 1;
+
+    assert!(
+        OpeningById::<T, I>::contains_key(opening_id),
+        "Opening not added"
+    );
+
+    opening_id
+}
+
+fn apply_on_opening_helper<T: Trait + working_group::Trait<I>, I: Instance>(
+    applicant_account_id: &T::AccountId,
+    applicant_member_id: &T::MemberId,
+    opening_id: &OpeningId,
+) -> ApplicationId {
+    working_group::Module::<T, I>::apply_on_opening(
+        RawOrigin::Signed((*applicant_account_id).clone()).into(),
+        ApplyOnOpeningParameters::<T> {
+            member_id: *applicant_member_id,
+            opening_id: *opening_id,
+            role_account_id: applicant_account_id.clone(),
+            reward_account_id: applicant_account_id.clone(),
+            description: vec![],
+            stake_parameters: StakeParameters {
+                stake: <T as working_group::Trait<I>>::MinimumApplicationStake::get(),
+                staking_account_id: applicant_account_id.clone(),
+            },
+        },
+    )
+    .unwrap();
+
+    let application_id = working_group::Module::<T, I>::next_application_id() - 1;
+
+    assert!(
+        ApplicationById::<T, I>::contains_key(application_id),
+        "Application not added"
+    );
+
+    application_id
+}
+
+fn create_storage_bucket_helper<T: Trait>(account_id: T::AccountId) -> T::StorageBucketId {
+    let storage_bucket_id = Module::<T>::next_storage_bucket_id();
+
+    Module::<T>::create_storage_bucket(RawOrigin::Signed(account_id).into(), None, true, 0, 0)
+        .unwrap();
+
+    assert!(<StorageBucketById<T>>::contains_key(&storage_bucket_id));
+
+    storage_bucket_id
+}
+
+fn create_storage_buckets<T: Trait>(
+    account_id: T::AccountId,
+    i: u32,
+) -> BTreeSet<T::StorageBucketId> {
+    iter::repeat(())
+        .take(i.saturated_into())
+        .map(|_| create_storage_bucket_helper::<T>(account_id.clone()))
+        .collect::<_>()
+}
+
+fn create_cids(i: u32) -> BTreeSet<Cid> {
+    fn create_cid(i: u32) -> Cid {
+        let bytes = i.to_be_bytes();
+        let mut buffer = Vec::new();
+
+        // Total CID = 32 bytes
+        for _ in 0..8 {
+            buffer.append(&mut bytes.to_vec());
+        }
```

Comment:
This is an actual size of the CID (32 bytes hash).

Reply:
I understand, but user can provide a `Vec` of any arbitrary length as an extrinsic argument and this will not be accounted for by the weigh calculation function, which is a security issue.

Currently the weight/fee will be exactly the same no matter if one provides 1000 valid 32 bytes CIDs or 1000 "CIDs" of 100 KB each, yet the difference in processing time etc. will be huge between those.

Reply:
I would rather add CID-length checks for the `update_blacklist` extrinsic. cc @bedeho

Reply:
1. Why would it be that this has such an effect? My mental model is that basically # database operations dominate everything else in weight function.
2. I am fine with adding a bound if that resolves the issue.

Reply:
Added CID-checks here: https://github.com/Joystream/joystream/pull/3393/commits/6897b1d1c3b2aaabc0992ceb414116df51781ae1